presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -374,22 +407,104 @@ private JoinEnumerationResult getJoinSource(LinkedHashSet<PlanNode> nodes, List<
             return chooseJoinOrder(nodes, outputVariables);
         }
 
-        private static boolean isJoinEqualityCondition(RowExpression expression)
+        @VisibleForTesting
+        static JoinCondition extractJoinConditions(List<RowExpression> joinPredicates,
+                Set<VariableReferenceExpression> leftVariables,
+                Set<VariableReferenceExpression> rightVariables,
+                VariableAllocator variableAllocator)
         {
-            return expression instanceof CallExpression
-                    && ((CallExpression) expression).getDisplayName().equals(EQUAL.getFunctionName().getObjectName())
-                    && ((CallExpression) expression).getArguments().size() == 2
-                    && ((CallExpression) expression).getArguments().get(0) instanceof VariableReferenceExpression
-                    && ((CallExpression) expression).getArguments().get(1) instanceof VariableReferenceExpression;
+            ImmutableMap.Builder<VariableReferenceExpression, RowExpression> newLeftAssignments = ImmutableMap.builder();
+            ImmutableMap.Builder<VariableReferenceExpression, RowExpression> newRightAssignments = ImmutableMap.builder();
+
+            ImmutableList.Builder<EquiJoinClause> joinClauses = ImmutableList.builder();
+            ImmutableList.Builder<RowExpression> joinFilters = ImmutableList.builder();
+
+            for (RowExpression predicate : joinPredicates) {
+                if (predicate instanceof CallExpression
+                        && (((CallExpression) predicate).getDisplayName().equals(EQUAL.name())
```

Comment:
Use this util to check EQUAL function
https://github.com/prestodb/presto/blob/3ffb027999197165cde90393b57b01c63112f8ee/presto-main/src/main/java/com/facebook/presto/sql/relational/FunctionResolution.java#L337